internal/outputs/storage/storage_client.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+)
+
+type CloudStorageClient struct {
+	ctx    context.Context
+	client *storage.Client
+}
+
+func NewCloudStorageClient(ctx context.Context) (*CloudStorageClient, error) {
+
+	client, err := storage.NewClient(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &CloudStorageClient{
+		ctx:    ctx,
+		client: client,
+	}, nil
+}
+
+func NewCloudStorageClientWithCredentialsFile(ctx context.Context, credentialsFile string) (*CloudStorageClient, error) {
+
+	client, err := storage.NewClient(ctx, option.WithCredentialsFile(credentialsFile))
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &CloudStorageClient{
+		ctx:    ctx,
+		client: client,
+	}, nil
+}
+
+func (c *CloudStorageClient) BucketExists(bucketName string) bool {
+	_, err := c.client.Bucket(bucketName).Attrs(c.ctx)
+	return err == nil
+}
+
+func (c *CloudStorageClient) Write(bucketName, objectName string, content []byte) error {
+	w := c.client.Bucket(bucketName).Object(objectName).NewWriter(c.ctx)
+	w.Write(content)
```

Comment:
I'll check in both places to make sure, because according to docs, the close should be the definitive place for it:

`"Since writes happen asynchronously, Write may return a nil error even though the write failed (or will fail). Always use the error returned from Writer.Close to determine if the upload was successful."`